lib/koi_config/core.rb

```diff
@@ -0,0 +1,189 @@
+# SOURCE: https://github.com/danielsdeleo/deep_merge.git
+
+module DeepMerge
```

Comment:
Question: Is there a reason to duplicate all of this rather than just use the [official gem](https://rubygems.org/gems/deep_merge) which we've literally copied into ours?

Reply:
Rails also provides a [`deep_merge` extension](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/core_ext/hash/deep_merge.rb), but in the interest of simplicity, we could just add the gem...

Reply:
@sobakasu can we use the official deep_merge gem?

Reply:
I've reworked the code to use rails' built-in Hash#deep_merge method, with a block which merges arrays.